controllers/openstackdataplanenodeset_controller.go

```diff
@@ -259,28 +236,21 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 
 	// Reconcile BaremetalSet if required
 	if len(instance.Spec.BaremetalSetTemplate.BaremetalHosts) > 0 {
```

Comment:
This would never be true without BaremetalHosts in NodeSetSpec and BaremetalSet won't be reconciled.  As I mentioned earlier, BaremetalHosts have to be populated either in a webhook or here by changing BaremetalSetTemplate. That's why BaremetalSet is never reconciled[1] and the job is failing waiting for the provision server[2]. 

[1] https://logserver.rdoproject.org/03/303/f1c1af4f09f1bb64c617c1c14041d1a1809d3b67/github-check/dataplane-operator-crc-podified-edpm-baremetal/e35c2a7/ci-framework-data/logs/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-904dd5e829f352acbaefd55cb607e65919bde1b62604664c13a74aaae96a125f/namespaces/openstack-operators/pods/dataplane-operator-controller-manager-7c5d78fb77-jz5m8/logs/dataplane-operator-controller-manager-7c5d78fb77-jz5m8.log

[2] https://logserver.rdoproject.org/03/303/f1c1af4f09f1bb64c617c1c14041d1a1809d3b67/github-check/dataplane-operator-crc-podified-edpm-baremetal/e35c2a7/job-output.txt

Reply:
Moved this into the NodeSet controller:
https://github.com/bshephar/dataplane-operator/blob/crd-proposal/controllers/openstackdataplanenodeset_controller.go#L240-L251

But we probably want to do this with a webhook instead of having the controller update the user owned spec. I haven't touched on Webhooks yet in this PR though. I think we probably need to decide on a few more things we want to see in those webhooks (defaults, etc). Adding to the NodeSet controller for now should keep us moving.